File: src/libs/ThirdParty.VBSuppliers/Properties/ExpediaRapid/IExpediaRapidSettings.vb
Imports ThirdParty.Abstractions

Namespace ExpediaRapid

    Public Interface IExpediaRapidSettings
        ReadOnly Property ApiKey(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property Secret(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property Scheme(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property Host(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property SearchPath(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property SearchRequestBatchSize(ByVal tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
        ReadOnly Property CountryCode(ByVal tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property SalesChannel(ByVal tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property SalesEnvironment(ByVal tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property SortType(ByVal tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property RatePlanCount(ByVal tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
        ReadOnly Property PaymentTerms(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property PartnerPointOfSale(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property LanguageCode(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property UseGZIP(ByVal tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
        ReadOnly Property AllowCancellations(ByVal tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
        ReadOnly Property OffsetCancellationDays(ByVal tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
        ReadOnly Property UserAgent(ByVal tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property BillingTerms(ByVal tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property PlatformName(ByVal tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property RateOption(tpAttributeSearch As IThirdPartyAttributeSearch) As String
        ReadOnly Property ValidateAffiliateID(tpAttributeSearch As IThirdPartyAttributeSearch) As Boolean
    End Interface

End Namespace

File: src/libs/iVectorOne.Suppliers/ExpediaRapid/IExpediaRapidSettings.cs
namespace iVectorOne.Suppliers.ExpediaRapid
{
    public interface IExpediaRapidSettings
    {
        string APIKey(IThirdPartyAttributeSearch tpAttributeSearch);
        string Secret(IThirdPartyAttributeSearch tpAttributeSearch);
        string Scheme(IThirdPartyAttributeSearch tpAttributeSearch);
        string Host(IThirdPartyAttributeSearch tpAttributeSearch);
        string SearchURL(IThirdPartyAttributeSearch tpAttributeSearch);
        int HotelBatchLimit(IThirdPartyAttributeSearch tpAttributeSearch);
        string SourceMarket(IThirdPartyAttributeSearch tpAttributeSearch);
        string SalesChannel(IThirdPartyAttributeSearch tpAttributeSearch);
        string SalesEnvironment(IThirdPartyAttributeSearch tpAttributeSearch);
        string SortType(IThirdPartyAttributeSearch tpAttributeSearch);
        int RatePlanCount(IThirdPartyAttributeSearch tpAttributeSearch);
        string PaymentTerms(IThirdPartyAttributeSearch tpAttributeSearch);
        string PartnerPointOfSale(IThirdPartyAttributeSearch tpAttributeSearch);
        string LanguageCode(IThirdPartyAttributeSearch tpAttributeSearch);
        bool UseGZip(IThirdPartyAttributeSearch tpAttributeSearch);
        bool AllowCancellations(IThirdPartyAttributeSearch tpAttributeSearch);
        int OffsetCancellationDays(IThirdPartyAttributeSearch tpAttributeSearch);
        string UserAgent(IThirdPartyAttributeSearch tpAttributeSearch);
        string BillingTerms(IThirdPartyAttributeSearch tpAttributeSearch);
        string PlatformName(IThirdPartyAttributeSearch tpAttributeSearch);
        string RateOption(IThirdPartyAttributeSearch tpAttributeSearch);
        bool ValidateAffiliateID(IThirdPartyAttributeSearch tpAttributeSearch);
        string TermsAndConditionsLink(IThirdPartyAttributeSearch tpAttributeSearch);
    }
}